### DISCUSSION TITLE
"std::bad_cast" error

### DISCUSSION THREAD

**Discussion by EdSheltonUKAEA** [idaholab/moose#28764] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Usually MOOSE gives fairily helpful or better information when a model won't run because of a bug, typo in the code or similar error, specifying the file and line of code where the problem is.

But trying to run what is admittedly a fairly complicated model split across 6 input files I've now got just this:

> *** ERROR ***
> std::bad_cast
> 
> Abort(1) on node 7 (rank 7 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 1) - process 7

Any advice on how to locate and debug this?
Searching for "bad_cast" on the website I can find several references to Framework requirements, etc.  How do I translate that into finding the bug that's stopping my code from running?

**Comment by GiudGiud** [comment]
Hello

You ll want to use a debugger to find out what "cast" is failing.
A bad_cast error is (most likely here) a C++ error for casting a reference to an object to the wrong kind of object. 

Ideally, you would try to reproduce this in serial rather than debug in parallel
https://mooseframework.inl.gov/application_development/debugging.html

**Reply by GiudGiud** [reply]
Usually we catch those errors and give more context. This must be in a loop that is not encapsulated
